containerFruits/Box.java

```diff
@@ -0,0 +1,71 @@
+package containerFruits;
+
+import containerFruits.abstracts.Fruit;
+import containerFruits.interfaces.Weightable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class Box<T extends Fruit> implements Weightable{
+
+    private ArrayList<T> fruits;
+
+    public Box() {
+        fruits = new ArrayList<>();
+    }
+
+    public ArrayList<T> getFruits() {
+        return fruits;
+    }
+
+    public void setFruits(ArrayList<T> fruits) {
+        this.fruits = fruits;
+    }
+
+    public boolean addFruit(T fruit){
+        if (fruit==null){
+            throw new RuntimeException("Invalid input parameter");
+        }
+
+        fruits.add(fruit);
+        return true;
+    }
+
+    public boolean addFruits(T[] fruits){
+        if (fruits == null || fruits.length==0){
+            throw new RuntimeException("Invalid input parameter");
+        }
+
+        Collections.addAll(this.fruits, fruits);
+        return true;
+    }

```

Comment:
You should check each element type because you're not allowed to add fruit of the different types into the current box.